### PR TITLE
Fixes #101: Moved existing RefreshEndpoint logic to a new RefreshSupport class.

### LINKED ISSUE #101: Configuration refreshing depends on actuator
See original discussion https://github.com/spring-cloud/spring-cloud-config/issues/352.

@dsyer I agree with @kurtinaitis, I use the `RefreshEvent` which has a listener in commons that does
`RefreshEndpoint.refresh()`. I have to guard that `RefreshEndpoint` is a bean in consul and zookeeper for their watches to trigger a refresh.

`RefreshEndpoint.refresh()` does the following, could be extracted.

		Map<String, Object> before = extract(
				this.context.getEnvironment().getPropertySources());
		addConfigFilesToEnvironment();
		Set<String> keys = changes(before,
				extract(this.context.getEnvironment().getPropertySources())).keySet();
		this.context.publishEvent(new EnvironmentChangeEvent(keys));
		this.scope.refreshAll();
		return keys.toArray(new String[keys.size()]);

### PR DESCRIPTION
**Pull request by venilnoronha**
Hi,

I've moved RefreshEndpoint's logic to RefreshSupport and delegated the _refresh_ call to it. One can now use RefreshSupport outside actuator. Please verify and merge.

My ICLA Number: 149720151120072904

Thanks,
Venil Noronha

### CODE REVIEW

File: spring-cloud-context/src/main/java/org/springframework/cloud/context/refresh/RefreshSupport.java [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+import org.springframework.core.env.MutablePropertySources;
+import org.springframework.core.env.PropertySource;
+import org.springframework.core.env.StandardEnvironment;
+import org.springframework.web.context.support.StandardServletEnvironment;
+
+/**
+ * 
+ * @author Venil Noronha
+ */
+public class RefreshSupport {
```
Comment (spencergibb) [inline spring-cloud-context/src/main/java/org/springframework/cloud/context/refresh/RefreshSupport.java]: Not so keen on the name. Maybe `EnvironmentRefresher`? @dsyer?

File: spring-cloud-context/src/main/java/org/springframework/cloud/autoconfigure/RefreshEndpointAutoConfiguration.java [hunk 2]
```diff
@@ -112,9 +117,15 @@ public RestartEndpoint restartEndpoint() {
 
 		@Bean
 		@ConditionalOnMissingBean
-		public RefreshEndpoint refreshEndpoint(ConfigurableApplicationContext context,
+		public RefreshSupport refreshSupport(ConfigurableApplicationContext context,
```
Comment (spencergibb) [inline spring-cloud-context/src/main/java/org/springframework/cloud/autoconfigure/RefreshEndpointAutoConfiguration.java]: This bean is still guarded by `@ConditionalOnClass(Endpoint.class)`. It should be reworked so it is not.

Comment (spencergibb) [inline spring-cloud-context/src/main/java/org/springframework/cloud/autoconfigure/RefreshEndpointAutoConfiguration.java]: It should probably go in `RefreshAutoConfiguration`.

File: spring-cloud-context/src/main/java/org/springframework/cloud/context/refresh/RefreshSupport.java (hunk 1 above)
Comment (dsyer) [inline spring-cloud-context/src/main/java/org/springframework/cloud/context/refresh/RefreshSupport.java]: Yeah, need a better name. We already have `EnvironmentManager`. What about `ContextRefresher`? I like the way that `ApplicationEvents` provide an extensibility already, though (you can listen for the event if you want to refresh more stuff). Maybe the `ContextRefresher` could just emit an event, and all the other features can be listeners?

Comment (spencergibb) [inline spring-cloud-context/src/main/java/org/springframework/cloud/context/refresh/RefreshSupport.java]: And log the `keys` as a different type of event?

Comment (venilnoronha) [inline spring-cloud-context/src/main/java/org/springframework/cloud/context/refresh/RefreshSupport.java]: I've renamed the class to `ContextRefresher`. I didn't get the `ApplicationEvents` part. Are we talking about publishing a new event along with `EnvironmentChangeEvent`?

File: spring-cloud-context/src/main/java/org/springframework/cloud/autoconfigure/RefreshEndpointAutoConfiguration.java (hunk 2 above)
Comment (venilnoronha) [inline spring-cloud-context/src/main/java/org/springframework/cloud/autoconfigure/RefreshEndpointAutoConfiguration.java]: I've moved it to `RefreshAutoConfiguration`.

**Comment by venilnoronha** [comment]
@spencergibb I've done the necessary changes. Could you please review? Also, do guide me with the `ApplicationEvents` change if need be.

**Comment by spencergibb** [comment]
@dsyer is this ok after we've gone into RC mode?

**Comment by dsyer** [comment]
Fine I think (it's not really a new feature).

**Comment by spencergibb** [comment]
Squashed and merged via 1e71c5632c01736fbe5be3b399318de23f0f0c3a

**Comment by venilnoronha** [comment]
Thank you!

**Pull request closed without merging**